helix-term/src/application.rs

```diff
@@ -867,7 +867,7 @@ impl Application {
                         // Insert the original lsp::Diagnostics here because we may have no open document
                         // for diagnosic message and so we can't calculate the exact position.
                         // When using them later in the diagnostics picker, we calculate them on-demand.
-                        match self.editor.diagnostics.entry(params.uri) {
+                        match self.editor.diagnostics.entry(path) {
```

Comment:
The path is not normalized here so this could still lead to inconsistencies/problems. We should normalize the path. I can't make a comment to the relative code since it's unchanged. Just call normalize functuon from helix core lm the result of `params.uri.to_file_path()` above. We already do this for other paths we receive from the LSP too.